src/components/IncorporationAgreement/AgreementType.vue

```diff
@@ -96,22 +96,23 @@ export default class AgreementType extends Vue {
   private showAgreementTypeForm = false
 
   mounted (): void {
-    this.setIncorporationAgreementStepData({
-      valid: !!this.agreementType,
-      agreementType: this.agreementType
-    })
-
+    this.setIncorporationAgreementData()
     // now that all data is loaded, wait for things to stabilize and reset flag
     Vue.nextTick(() => this.setHaveChanges(false))
```

Comment:
The global haveChanges flag was to prevent a user from exiting with unsaved changes, but it got complicated and I think it's not working correctly -- for example, does it currently identify unsaved changes, or any change from the original IA?

I think the best thing is to initialize this flag (to false) when the Correction view has finished loading, and then set it true when any data changes (in mutations-model.ts). (Some mutations already set the flag to true.)

Setting this flag in this component is not needed. (This code is left over from Create UI.)

Reply:
Agree that it is overly complicated and very difficult to understand. I have removed that from component. Also added the mutation in setNameRequest which was not present before. I think the functionality works as expected now. Please confirm

Reply:
Resolving this conversation... I'll look at your updated code ASAP.